src/LinuxTracing/LinuxTracingIntegrationTest.cpp

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/strings/numbers.h>
+#include <gtest/gtest.h>
+#include <stdint.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <array>
+#include <chrono>
+#include <thread>
+
+#include "LinuxTracing/Tracer.h"
+#include "LinuxTracing/TracerListener.h"
+#include "LinuxTracingIntegrationTestPuppet.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ReadFileToString.h"
+#include "capture.pb.h"
+
+namespace orbit_linux_tracing {
+namespace {
+
+[[nodiscard]] bool IsRunningAsRoot() { return geteuid() == 0; }
+
+[[nodiscard]] __attribute__((unused)) bool CheckIsRunningAsRoot() {
+  if (IsRunningAsRoot()) {
+    return true;
+  }
+
+  ERROR("Not running as root");
+  return false;
+}
+
+[[nodiscard]] int ReadPerfEventParanoid() {
+  auto error_or_content = orbit_base::ReadFileToString("/proc/sys/kernel/perf_event_paranoid");
+  CHECK(error_or_content.has_value());
+  const std::string& content = error_or_content.value();
+  int perf_event_paranoid = 2;
+  bool atoi_succeeded = absl::SimpleAtoi(content, &perf_event_paranoid);
+  CHECK(atoi_succeeded);
+  return perf_event_paranoid;
+}
+
+[[nodiscard]] bool CheckIsPerfEventParanoidAtMost(int max_perf_event_paranoid) {
+  if (IsRunningAsRoot()) {
+    return true;
+  }
+
+  int perf_event_paranoid = ReadPerfEventParanoid();
```

Comment:
I can see the value for running locally - but once we have tests for uprobes paranoid will not be enough afaik.

Reply:
That's why I added `[[nodiscard]] __attribute__((unused)) bool CheckIsRunningAsRoot()` above, I will use that for uprobes, but I want each test to have the exact condition.